bay/images/Dockerfile.node

```diff
@@ -0,0 +1,42 @@
+FROM uselagoon/node-14:20.12.0
+
+ENV CYPRESS_INSTALL_BINARY 0
+ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD 1
+
+# Build time env vars
+ONBUILD ARG CONTENT_API_SERVER
+ONBUILD ARG CONTENT_API_AUTH_PASS
+ONBUILD ARG CONTENT_API_AUTH_USER
+ONBUILD ARG CONTENT_API_OAUTH_CLIENT_ID
+ONBUILD ARG BASIC_AUTH
+ONBUILD ARG TIDE_DEBUG
+ONBUILD ARG DISPLAY_ERROR
+ONBUILD ARG GTM_ID
+ONBUILD ARG SEARCH_HASH
+ONBUILD ARG SEARCH_SERVICE
+ONBUILD ARG SEARCH_INDEX
+ONBUILD ARG SEARCH_URL
+ONBUILD ARG SEARCH_LOG
+ONBUILD ARG SEARCH_AUTH_USERNAME
+ONBUILD ARG SEARCH_AUTH_PASSWORD
+ONBUILD ARG LAGOON_ENVIRONMENT_TYPE
+
+ONBUILD ENV CONTENT_API_SERVER=$CONTENT_API_SERVER
+ONBUILD ENV CONTENT_API_AUTH_PASS=$CONTENT_API_AUTH_PASS
+ONBUILD ENV CONTENT_API_AUTH_USER=$CONTENT_API_AUTH_USER
+ONBUILD ENV CONTENT_API_OAUTH_CLIENT_ID=$CONTENT_API_OAUTH_CLIENT_ID
+ONBUILD ENV BASIC_AUTH=$BASIC_AUTH
+ONBUILD ENV TIDE_DEBUG=$TIDE_DEBUG
+ONBUILD ENV DISPLAY_ERROR=$DISPLAY_ERROR
+ONBUILD ENV GTM_ID=$GTM_ID
+ONBUILD ENV SEARCH_HASH=$SEARCH_HASH
+ONBUILD ENV SEARCH_SERVICE=$SEARCH_SERVICE
+ONBUILD ENV SEARCH_INDEX=$SEARCH_INDEX
+ONBUILD ENV SEARCH_URL=$SEARCH_URL
+ONBUILD ENV SEARCH_LOG=$SEARCH_LOG
+ONBUILD ENV SEARCH_AUTH_USERNAME=$SEARCH_AUTH_USERNAME
+ONBUILD ENV SEARCH_AUTH_PASSWORD=$SEARCH_AUTH_PASSWORD
+ONBUILD ENV LAGOON_ENVIRONMENT_TYPE=$LAGOON_ENVIRONMENT_TYPE
```

Comment:
On further thought, do we really need these values as runtime environment variables? 

Based on the conversation with Tim last week, the values are read during the build and that is it.

Reply:
Just to clarify `ARG` will set values during the build of the image and `ENV` will persist in the resulting image.

@tim-yao If the application is compiled during the build is there any requirement for the resulting image to have environment variables defined? I'm thinking it may be a requirement if there's some SSR going on?

Reply:
@GROwen @nicksantamaria Yes, not really need them as I remembered. But they are always there in Lagoon runtime so  it's possible has one or two be used by some code in runtime. As we just finished the regression test on this image, I think at this stage let's just keep them. We can remove them in another ticket and do another regression test before release the change.

Reply:
That never happens though (retrospective removal). I vote for cleaning up the cruft now - assuming it is cruft 😬.

I've created a new Docker Hub repo to set the `ripple-node` namespace. I'll remove the `ENV` commands and let that one build.

Reply:
@tim-yao As discussed in Slack I'm moving the discussion here to loop in @steveworley to help confirm inheritance of vars.

@steveworley Is it right that the vars we have defined as `ONBUILD ENV` are going in the container anyway because of Lagoon injecting it's env vars/configmap? If that's true then there will be zero impact of removing the `ONBUILD ENV` cmds.

Reply:
>If that's true then there will be zero impact of removing the ONBUILD ENV cmds.

This is correct. Values in the configmap will override anything set in the image.

Reply:
This just means we have to define the variables in lagoon with a `GLOBAL` scope (available both at runtime and at build time).

I'm pretty sure when they have a `BUILD` scope they don't get injected into the configmap and then are not available in the running containers, so we either have `ONBUILD ENV` or define variables with global scope in lagoon.